client/update_resumer.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2017 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package client
+
+import (
+	"fmt"
+	"github.com/mendersoftware/log"
+	"github.com/pkg/errors"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type UpdateResumer struct {
+	stream        io.ReadCloser
+	api           ApiRequester
```

Comment:
I would change the name to requester or similar. API is too broad IMHO.

Reply:
Yep, agree. Will change!